app/src/main/java/com/sun/mangareader01/ui/adapter/TrendingPagerAdapter.kt

```diff
@@ -0,0 +1,91 @@
+package com.sun.mangareader01.ui.adapter
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.FragmentManager
+import androidx.fragment.app.FragmentStatePagerAdapter
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout
+import com.sun.mangareader01.R
+import com.sun.mangareader01.data.model.Manga
+import com.sun.mangareader01.ui.listener.OnItemClickListener
+import kotlinx.android.synthetic.main.fragment_trending_pager.layoutRefresh
+import kotlinx.android.synthetic.main.fragment_trending_pager.recyclerMangas
+
+class TrendingPagerAdapter(
+    fragmentManager: FragmentManager,
+    private val fragments: List<PagerFragment>
+) : FragmentStatePagerAdapter(
+    fragmentManager,
+    BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT
+) {
+
+    override fun getItem(position: Int) = fragments[position]
+
+    override fun getCount() = fragments.size
+
+    override fun getPageTitle(position: Int) = getItem(position).title
+
+    class PagerFragment : Fragment() {
+
+        var title: String? = null
+        var icon: Int? = null
+        private var clickListener: OnItemClickListener? = null
+        private var refreshListener: SwipeRefreshLayout.OnRefreshListener? =
+            null
+
+        private val mangaAdapter: CustomAdapter<Manga> by lazy {
+            MangaAdapter(mutableListOf())
+        }
+
+        override fun onCreateView(
+            inflater: LayoutInflater,
+            container: ViewGroup?,
+            savedInstanceState: Bundle?
+        ): View? =
+            inflater.inflate(R.layout.fragment_trending_pager, container, false)
+
+        override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+            super.onViewCreated(view, savedInstanceState)
+            displayLoadingBar()
+            recyclerMangas.apply {
+                layoutManager = LinearLayoutManager(context)
+                adapter = mangaAdapter as RecyclerView.Adapter<*>
+            }
+            mangaAdapter.onItemClickListener = clickListener
+            layoutRefresh.setOnRefreshListener(refreshListener)
+        }
+
+        fun updateData(data: List<Manga>) {
+            mangaAdapter.updateData(data)
+            hideLoadingBar()
+        }
+
+        private fun displayLoadingBar() {
+            layoutRefresh?.isRefreshing = true
+        }
+
+        private fun hideLoadingBar() {
+            layoutRefresh?.isRefreshing = false
+        }
+
+        companion object {
+            @JvmStatic
+            fun newInstance(
+                title: String,
+                icon: Int,
+                clickListener: OnItemClickListener?,
+                refreshListener: SwipeRefreshLayout.OnRefreshListener
+            ) = PagerFragment().apply {
```

Comment:
e check convention 2.11
https://github.com/framgia/coding-standards/blob/master/eng/android/codingstyleguide.md#211-argument-in-activity-and-fragment